src/lib/FreeC/Backend/Coq/Analysis/DecreasingArguments.hs

```diff
@@ -232,6 +232,11 @@ checkDecArgs decls knownDecArgIndecies decArgIndecies = all
       let smaller' = withoutArgs varPats smaller
       in  checkExpr decArg smaller' expr []
 
+    checkBind :: IR.Bind -> Bool
+    checkBind (IR.Bind _ varPat expr) =
+      let smaller' = withoutArgs [varPat] smaller
```

Comment:
All bound variables are visible in all binders. Thus, you have to use the same `smaller'` as in the `IR.Let` rule of `checkExpr'`.